### PR TITLE
Finished Viscomm Recruit Training

### PR DESCRIPTION
**Pull request by Ahmed-I-Abdullah**

### CODE REVIEW

File: Recruit-Training/Viscomm-Recruit-Training/src/BatteryStateDisplayService.cpp [hunk 1]
```diff
@@ -25,9 +25,11 @@ void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& ba
 {
     batteryStateOfChargeService_.addData(batteryData);
 
+
     QTextStream(stdout) << "Voltage: " << batteryData.voltage
-        << " Current: " << batteryData.current
-        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed() << endl;
+                        << " Current: " << batteryData.current
+                        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed()
+                        << " Time Till Charged or Depleted: " << batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString() << endl;
```
Comment (DatLam25) [inline Recruit-Training/Viscomm-Recruit-Training/src/BatteryStateDisplayService.cpp:32]: Time should be printed out as HH:MM:SS.ZZZ.

File: Recruit-Training/Viscomm-Recruit-Training/src/BatteryStateOfChargeService.cpp [hunk 2]
```diff
@@ -7,31 +8,75 @@ namespace
...
+}
+
+void BatteryStateOfChargeService::updateAmpHours(){
+
+    if(pastData_->time.QTime::isNull()){
+        return;
+    }
+
+    double currentAvg = totalCurrent_ / currentCount_;
+    double timeDifference = pastData_->time.QTime::msecsTo(currentData_->time) / 3600000.0;
```
Comment (DatLam25) [inline Recruit-Training/Viscomm-Recruit-Training/src/BatteryStateOfChargeService.cpp:32]: All instances of `3600000.0` should be replaced with a constant so we can know that the number means.

**Review by Hesham-E — COMMENTED** [review]
Hey this looks great to me! You've got the hang of this so that's awesome. The only thing that I want to mention is that we follow a style guide when programming so that our code is structrued uniformly throughout. For example, we format our if statements like this

if (condition)
{
    code to run
}

This link will explain what you need to know. https://github.com/UCSolarCarTeam/Development-Resources/tree/master/Development-Guide/Style-Guide

**Comment by DatLam25** [comment]
This looks good to me. Well done on finishing the training. Wait for another review before closing this PR.

**Review by DatLam25 — APPROVED** [review]

**Review by Hesham-E — CHANGES_REQUESTED** [review]
Looks good! (Whoops I meant to do this as an approve)

**Review by Hesham-E — APPROVED** [review]
There we go

**Pull request closed without merging**